src/ui/public/stringify/types/color.js

```diff
@@ -1,53 +1,55 @@
 import _ from 'lodash';
+import { asPrettyString } from 'ui/utils/as_pretty_string';
 import { DEFAULT_COLOR } from './color_default';
 import { FieldFormat } from 'ui/index_patterns/_field_format/field_format';
 
 export function stringifyColor() {
 
   const convertTemplate = _.template('<span style="<%- style %>"><%- val %></span>');
 
-  _.class(_Color).inherits(FieldFormat);
-  function _Color(params) {
-    _Color.Super.call(this, params);
+  class ColorFormat extends FieldFormat {
+    constructor(params) {
+      super(params);
+    }
+
+    getParamDefaults() {
+      return {
+        fieldType: null, // populated by editor, see controller below
+        colors: [_.cloneDeep(DEFAULT_COLOR)]
+      };
+    }
+
+    findColorRuleForVal(val) {
+      switch (this.param('fieldType')) {
+        case 'string':
+          return _.findLast(this.param('colors'), (colorParam) => {
+            return new RegExp(colorParam.regex).test(val);
+          });
+
+        case 'number':
+          return _.findLast(this.param('colors'), ({ range }) => {
+            if (!range) return;
+            const [start, end] = range.split(':');
+            return val >= Number(start) && val <= Number(end);
+          });
+
+        default:
+          return null;
+      }
+    }
   }
 
-  _Color.id = 'color';
-  _Color.title = 'Color';
-  _Color.fieldType = [
+  ColorFormat.id = 'color';
+  ColorFormat.title = 'Color';
+  ColorFormat.fieldType = [
     'number',
     'string'
   ];
 
-  _Color.prototype.getParamDefaults = function () {
-    return {
-      fieldType: null, // populated by editor, see controller below
-      colors: [_.cloneDeep(DEFAULT_COLOR)]
-    };
-  };
-
-  _Color.prototype.findColorRuleForVal = function (val) {
-    switch (this.param('fieldType')) {
-      case 'string':
-        return _.findLast(this.param('colors'), (colorParam) => {
-          return new RegExp(colorParam.regex).test(val);
-        });
-
-      case 'number':
-        return _.findLast(this.param('colors'), ({ range }) => {
-          if (!range) return;
-          const [start, end] = range.split(':');
-          return val >= Number(start) && val <= Number(end);
-        });
-
-      default:
-        return null;
-    }
-  };
-
-  _Color.prototype._convert = {
+  ColorFormat.prototype._convert = {
```

Comment:
Actually, we only create an instance of the field formatter once per index pattern, there won't be any issue if you use instance variables for this. It would probably make things a little more readable as well.